src/main/scala/strawman/collection/Set.scala

```diff
@@ -1,4 +1,27 @@
 package strawman
 package collection
 
-trait Set[A] extends Iterable[A] {}
+import scala.Boolean
+
+/** Base trait for set collections */
+trait Set[A]
+  extends Iterable[A]
+    with SetLike[A, Set]
+
+/** Base trait for set operations */
+trait SetLike[A, +C[X] <: Set[X]]
+  extends IterableLike[A, C]
```

Comment:
Coming back to my argument for `StableIterable` from the other PR, we also inherit methods like `take` and `drop` from `Iterable` which only make sense with a stable ordering. Should we move them down to `StableIterable`? Or is this not important enough in practice and we should simply allow it for non-stable orderings, too (like in 2.12 and the current strawman design)?